src/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -169,8 +169,6 @@ auto PenInputHandler::actionMotion(InputEvent const& event) -> bool {
     GtkXournal* xournal = this->inputContext->getXournal();
     ToolHandler* toolHandler = this->inputContext->getToolHandler();
 
-    this->changeTool(event);
```

Comment:
I agree it seems weird that `changeTool` would be called as part of this function.
Here, I would be a little more cautious though because I'd assume someone put it there for a reason.

Maybe it would be good to have this as a separate mini-PR so multiple people can check it on their system if they experience any weird behaviour. Performance-wise it would be quite an improvement. Good catch :+1:

Reply:
That is actually on purpose. Some stylised do not send ENTER or LEAVE events for some reason. For those we must catch the motion events to handle the necessary updates when a pointer enters the drawing area. Also moving from background to a page in the drawing area must update the pointer which was so far handled by this call. 

There might be room to improve this nowadays. That code is at least two years old.